eventstore/test_whatsapp_actions.py

```diff
@@ -216,6 +216,36 @@ def test_handle_edd_message(self):
 
 
 class HandleEventTests(DjangoTestCase):
+    @override_settings(RAPIDPRO_OPTOUT_FLOW="test-flow-uuid")
+    def test_whatsapp_delivery_failure_error(self):
+        """
+        If the event is of type Failed, and uses whatsapp,
+        then it should trigger the message delivery failed action
+        """
+        event = Event.objects.create()
+        event.fallback_channel = False
+        event.status = Event.FAILED
+        event.recipient_id = "27820001001"
+        event.timestamp = timezone.now() + timedelta(days=2)
+
+        DeliveryFailure.objects.create(number_of_failures=4, contact_id="27820001001")
+
+        with patch("eventstore.tasks.rapidpro") as p:
+            handle_event(event)
+
+        p.create_flow_start.assert_called_once_with(
+            extra={
+                "optout_reason": "whatsapp_failure",
```

Comment:
Just noting that we must remember to update the flow as well, to handle this new reason (it needs to be mapped to a number to submit to openhim)

Reply:
awesome, thanks!